### ISSUE TITLE
a11y: improve input group accessibility

### ISSUE THREAD

**Issue by desig9stein** [IgniteUI/igniteui-angular#9996]
## Description  

Provide an accessible experience for igx-input-group.

~~1. If the input-group has a label, it should automatically be used as the `aria-label` for the input filed~~
~~2. if the input-group has `hint` it should be added to the `aria-describedby` list of the input field~~
~~3. `aria-invalid` should be added and automatically updated based on the input's validity state.~~
~~4. `aria-required` should be added if the input is a required field.~~
~~5. To notify the user that there is a problem, Add `role="alert"` to the error message.~~

~~NOTE:  According to [MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/forms/alerts) `aria-required` is required only for browsers that do not yet support ARIA.~~

1. `aria-invalid` should be added and automatically updated based on the input's validity state.
2. `aria-required` should be added if the input is a required field. This should cover the case where the input group is used within a reactive form. With a native `required` applied to the input or with the `igxInput` directive's `required` property set to `true` there's no need to explicitly add the `aria-required` attribute.

## Attachments  
Attach a sample if available, and screenshots, if applicable.
